allennlp/training/trainer.py

```diff
@@ -284,10 +284,18 @@ def _metrics_to_tensorboard(self,
         """
         Sends all of the train metrics (and validation metrics, if provided) to tensorboard.
         """
-        for name, value in train_metrics.items():
-            self._tensorboard.add_train_scalar(name, value, epoch)
-            if val_metrics:
-                self._tensorboard.add_validation_scalar(name, val_metrics[name], epoch)
+
+        metric_names = set(train_metrics.keys())
+        if val_metrics:
+            metric_names.update(val_metrics.keys())
+
+        for name in metric_names:
+            train_metric = train_metrics.get(name, None)
```

Comment:
the second argument to `get` is `None` by default, so you don't need to specify it

Reply:
this one too, and also in line 296